python-lib/config.py

```diff
@@ -121,8 +123,7 @@ def get_snowflake_to_hdfs_query(sf_location: AnyStr, sf_table_name: AnyStr,
     # This is required as the COPY command does not support TZ and LTZ
     timezoned_types = ['TIMESTAMPLTZ', 'TIMESTAMPTZ', 'TIMESTAMP']
     columns = [
-        f'"{c["name"]}"'
-        + (f'::TIMESTAMP_NTZ AS "{c["name"]}"' if c['originalType'] in timezoned_types else '')
+        f'"{c["name"]}"' + (f'::TIMESTAMP_NTZ AS "{c["name"]}"' if c['originalType'] in timezoned_types else '')
```

Comment:
This change is going to be in conflict with my PR. No need to handle TIMESTAMP type as internally it is replaced by actual type. As per [Snowflake documentation] (https://docs.snowflake.com/en/sql-reference/data-types-datetime.html) 'The TIMESTAMP data type is never stored in tables.'

Reply:
IIRC, `['TIMESTAMPLTZ', 'TIMESTAMPTZ', 'TIMESTAMP']` were the observed data types from `read_schema()` in the instances where `read_schema` was actually populating `originalType`. Because this method is still acting upon what it'd get from `read_schema`, it should continue to cover the cases where it could theoretically end up getting `TIMESTAMP`. In other words: this method doesn't know that we've manually retrieved an `originalType` value directly from Snowflake.

Reply:
I do not know where the requirement to support Dataiku schema in this method is coming from. I understand that originally it was using one, but AFAIK this method is only used in one place so not sure what we are trying to achieve here.

Reply:
We require clean, consistent code. Consumers should be able to expect that two methods with the same signature would expect the same input (a dataiku schema).

Passing in a list of columns and snowflake types would probably be fine, too. It'd mean the method would have a different signature and no ambiguity about what `originalType` actually means.

Reply:
Ok, in order to be consistent we need to be 100% sure that `REPLACE(data_type, '_', '') AS "originalType"` will be matching to Dataiku datatype for all Snowflake types. I currently do not have this confidence, so would prefer just using Snowflake data types as is. With regards to method signature - it is not specific to Dataiku imho.

Reply:
RE https://github.com/outrauk/dataiku-plugin-snowflake-hdfs/pull/17#issuecomment-666291831

Okay, understood.

Two options, which do you prefer?

1. Map the additional types (`TEXT` &rarr; `VARCHAR`, `FLOAT` &rarr; `DOUBLE`, and assume there aren't other differences)
2. Pass this as `{"name": "someColumnName", "sfDataType": "TIMESTAMP_NTZ"}` and add the conditional for `sfDataType` in `get_snowflake_to_hdfs_query`

Reply:
I went with #2 which allowed me to `git cherry-pick` your PR's commit.